### ISSUE TITLE
C++ API

### ISSUE THREAD

**Issue by subr3v** [TachTech/Horde3D#5]
GIST Here:
https://gist.github.com/subr3v/09e82199c9d93e986623

**Comment by subr3v** [comment]
Draft for new API:

class Engine
{
	public:
		void Init();
		void Shutdown();
		const char* GetVersion();
		ErrorManager& GetErrorManager();
		
};

struct ErrorMessage
{
	const char* Error;
	int Level;
	float Time;
};

class ErrorManager
{
	public:
		bool HasErrors();
		void GetError(ErrorMessage& OutError);
};

**Comment by MarkSkyzoid** [comment]
Be careful with the resource class, as there is already an Horde3D::Resource class declared here:
https://github.com/TachTech/Horde3D/blob/master/Horde3D/Source/Horde3DEngine/egResource.h

I would suggest going for something else or hide the horde3d implementation completely, as that might create confusion.

Also, in your Resource class I do not like the redundancy of the names such as GetResourceType() or GetResourceName(). They are both member functions of the Resource class, thus the Resource in the names can be avoided.

**Comment by subr3v** [comment]
The idea is to hide the horde3D implementation completely, so you would not be able to access any of those internal classes. As for the redundancy, you're right but at the same time we might clash with stuff such as Textures which might require a GetType method (and then it would be a redundant GetTextureType), so it's a matter of tradeoff there, we need to decide whether it's worth renaming it to something else

**Comment by MarkSkyzoid** [comment]
In the first  case, it might be worth creating a new namespace. Thus, decide a name for the middleware and be consistent with it. 
If a Texture inherits from a Resource base class, I still think offering a generic accessor name is our best bet.

**Comment by subr3v** [comment]
Yep, of course. The API is there just to give the general idea, we should find a name for the "new" engine and make a namespace as well. I guess we can change the GetResourceType to GetType and so on, we'll find a solution for specific problems such as the Texture one later on.
